fox_scraper/fox_scraper.py

```diff
@@ -48,63 +72,54 @@ def __init__(self, site_address):
     def get_web_content(self, http_address):
         """
         :param http_address: url address to send a request
-        :return: an html text to be scrap from
+        :return: an html parse text to be scrap from (beutifulsoup object)
         """
         raw_data = requests.get(http_address)
-        return raw_data.text
+        return BeautifulSoup(raw_data.text, 'html.parser')
 
-    def get_soup_object(self, web_text):
-        """
-        :param web_text: html txt file to scrap
-        :return: beautiful soup objet for scraping
-        """
-        return BeautifulSoup(web_text, 'html.parser')
-
-    def scrap_fox_runner(self):
+    def runner(self):
         """
         one of the main function for this class.
         this function will activate the scraping of the web address revieced on contruction of the class
         :return: a list of items (Item class) of the entire catalog
         """
-        soup_main_categories = self.scrap_fox_main_categories()
+        soup_main_categories = self.get_main_categories()
         catalog = []
-        soup_sub_categories = []
-        main_cat_num = 0
-        for soup_main_category in soup_main_categories:
-            soup_sub_categories += list(self.scrap_fox_sub_categories(soup_main_category, main_cat_num))
-            catalog.append(self.scrap_fox_catalog(soup_sub_categories, main_cat_num))
-            main_cat_num += 1
+        for main_cat_num in range(len(soup_main_categories)): # for every main-category
+            soup_sub_categories = self.get_sub_categories(soup_main_categories[main_cat_num], main_cat_num)
+            catalog.append(self.get_full_catalog(soup_sub_categories, main_cat_num))
```

Comment:
this line `catalog.append(self.get_full_catalog(soup_sub_categories, main_cat_num))` to
```
sub_catalog = self.get_full_catalog(soup_sub_categories, main_cat_num)
catalog.append(sub_catalog)
```